File: src/common/game/main/java/settingdust/mod_sets/game/util/ModSetLoadCallback.java
package settingdust.mod_sets.game.util;


import settingdust.mod_sets.util.Event;

import java.util.List;

public class ModSetLoadCallback {
    public static Event<Callback> EVENT = new Event<>((List<Callback> listeners) ->
        () -> {
            for (Callback listener : listeners) {
                listener.onLoad();
            }
        }
    );

    @FunctionalInterface
    public interface Callback {
        void onLoad();
    }
}

File: src/common/game/main/kotlin/settingdust/mod_sets/game/util/ModSetLoadCallback.kt
package settingdust.mod_sets.game.util

import settingdust.kinecraft.event.Event

object ModSetLoadCallback {
    @JvmField
    val CALLBACK = Event<Callback> { listeners ->
        Callback {
            for (callback in listeners) {
                callback.onModSetLoad()
            }
        }
    }

    fun interface Callback {
        fun onModSetLoad()
    }
}
